gen/src/reader/utils.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+
+
+// TODO: Get source ID and call download for file path?
```

Comment:
Unless we want to parallelize the generation as well, doing download and generate in two separate steps is probably the best idea. The parallelization of downloads that Hyper allows to happen is why the downloads go almost sillily quickly.

Reply:
This one was just about not hardcoding paths and instead calling the `download` code to give us the path, having the source name, test status, and file name. But I don't think we're going down that path, as keeping the download script completely separate looks the best, for reasons I'm going to explain in a follow-up comment.